src/ObjectStore/v1/Models/Object.php

```diff
@@ -96,11 +96,14 @@ public function retrieve()
      * distinct from fetching its metadata (a `HEAD` request). The body of an object is not fetched by default to
      * improve performance when handling large objects.
      *
+     * @param array $options {@see \OpenStack\ObjectStore\v1\Api::getObject}
+     *
      * @return StreamInterface
      */
-    public function download(): StreamInterface
+    public function download(array $options = []): StreamInterface
     {
-        $response = $this->executeWithState($this->api->getObject());
+        $obj = $this->api->getObject();
+        $response = $this->execute($obj, $options + $this->getAttrs(array_keys($obj['params'])));
```

Comment:
@jamiehannaford  @haphan  OK, I agree with 

$response = $this->execute($this->api-> getObject(), $data + ['containerName' => $this->containerName])

BTW since it's minor change, can @haphan  make this final change? Otherwise, I have to play around with php-cs-fixer for a while.